plugins/obs-ffmpeg/obs-ffmpeg-vaapi.c

```diff
@@ -511,18 +512,73 @@ static void set_visible(obs_properties_t *ppts, const char *name, bool visible)
 
 static void vaapi_defaults(obs_data_t *settings)
 {
-	obs_data_set_default_string(settings, "vaapi_device",
-				    "/dev/dri/renderD128");
+	const char *device = vaapi_get_h264_default_device();
+	VADisplay va_dpy;
+
+	obs_data_set_default_string(settings, "vaapi_device", device);
 	obs_data_set_default_int(settings, "profile",
 				 FF_PROFILE_H264_CONSTRAINED_BASELINE);
 	obs_data_set_default_int(settings, "level", 40);
 	obs_data_set_default_int(settings, "bitrate", 2500);
 	obs_data_set_default_int(settings, "keyint_sec", 0);
 	obs_data_set_default_int(settings, "bf", 0);
 	obs_data_set_default_int(settings, "rendermode", 0);
-	obs_data_set_default_string(settings, "rate_control", "CBR");
 	obs_data_set_default_int(settings, "qp", 20);
 	obs_data_set_default_int(settings, "maxrate", 0);
+
+	va_dpy = vaapi_open_device(device, "vaapi_defaults");
+
+	if (vaapi_device_rc_supported(VAProfileH264ConstrainedBaseline, va_dpy,
+				      VA_RC_CBR, device))
+		obs_data_set_default_string(settings, "rate_control", "CBR");
+	else if (vaapi_device_rc_supported(VAProfileH264ConstrainedBaseline,
+					   va_dpy, VA_RC_CQP, device))
+		obs_data_set_default_string(settings, "rate_control", "CQP");
+	else
+		obs_data_set_default_string(settings, "rate_control", "VBR");
```

Comment:
Why change the default mode to VBR from CBR? I can understand adding a fallback here for things like intel that only support CQP on free drivers, but VBR is just as unsupported as CBR on these drivers.

Reply:
We have no guarantee that a driver will provide CBR, VBR and CQP or only CQP.

It just avoid to put a wrong value as default.

Reply:
sorry i mean, if we are going to have an else branch and change it from CBR it seems the best choice is CQP not VBR. Since CQP is more supported on intel and amd just supports all 3 mostly?

Reply:
`else` is now CQP

Reply:
It was mistake from me for the order.